tests/unittests/tests-at/tests-at.c

```diff
@@ -0,0 +1,693 @@
+/*
+ * Copyright (C) 2024 ML!PA GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include "embUnit.h"
+
+#include "at.h"
+#include "isrpipe/read_timeout.h"
+
+#include "tests-at.h"
+
+#define UNIT_TEST_LONG_URC "+UNITTEST_LONG_URC_VEEERY_LONG"
+#define UNIT_TEST_SHORT_URC "+U"
+#define LONG_COMMAND "AT+COMMAND_AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"\
+    "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+#define COMMAND_WITH_EOL "AT+COMMAND_"AT_RECV_EOL"BLABLA"AT_RECV_EOL"BLA"
+
+at_dev_t at_dev;
+static char buf[256];
+static char rp_buf[256];
+#ifdef MODULE_AT_URC
+static unsigned urc_count = 0;
+
+void unit_test_urc_long_handler(void *arg, char const *code)
+{
+    TEST_ASSERT(strncmp(UNIT_TEST_LONG_URC, code, strlen(UNIT_TEST_LONG_URC)) == 0);
+    unsigned *urc_count = (unsigned *)arg;
+    *urc_count += 1;
+}
+
+void unit_test_urc_short_handler(void *arg, char const *code)
+{
+    TEST_ASSERT(strncmp(UNIT_TEST_SHORT_URC, code, strlen(UNIT_TEST_SHORT_URC)) == 0);
+    unsigned *urc_count = (unsigned *)arg;
+    *urc_count += 1;
+}
+
+at_urc_t urc_long = {
+    .arg = &urc_count,
+    .code = UNIT_TEST_LONG_URC,
+    .cb = unit_test_urc_long_handler
+};
+at_urc_t urc_short = {
+    .arg = &urc_count,
+    .code = UNIT_TEST_SHORT_URC,
+    .cb = unit_test_urc_short_handler
+};
+#endif
+
+static void set_up(void)
+{
+    at_dev_init_t at_init_params = {
+        .baudrate = 115200,
+        .rp_buf = rp_buf,
+        .rp_buf_size = sizeof(rp_buf),
+        .rx_buf = buf,
+        .rx_buf_size = sizeof(buf),
+        .uart = UART_DEV(0),
+    };
+    int res = at_dev_init(&at_dev, &at_init_params);
+    /* check the UART initialization return value and respond as needed */
+    if (res == UART_NODEV) {
+        TEST_FAIL("Invalid UART device given!");
+    }
+    if (res == UART_NOBAUD) {
+        TEST_FAIL("Baudrate is not applicable!");
+    }
+
+    /* we don't use the serial device, make sure it doesn't clobber our rx buffer */
+    at_dev_poweroff(&at_dev);
+    at_drain(&at_dev);
+
+#ifdef MODULE_AT_URC
+    at_add_urc(&at_dev, &urc_long);
+    at_add_urc(&at_dev, &urc_short);
+#endif
+}
+
+static void tear_down(void)
+{
+#ifdef MODULE_AT_URC
+    at_remove_urc(&at_dev, &urc_long);
+    at_remove_urc(&at_dev, &urc_short);
+#endif
+}
+
+static void assert_urc_count(unsigned expected)
+{
+#ifdef MODULE_AT_URC
+    TEST_ASSERT_EQUAL_INT(expected, urc_count);
+    urc_count = 0;
+#endif
+    (void)expected;
+}
+
+
```

Comment:
```suggestion

```

i am quiet sure that l.99 and l.100 are both just `\n` please remove one of them no one will jugde you for choosing the wrong one